src/main/java/org/jabref/gui/sidepane/SidePaneContainerViewModel.java

```diff
@@ -0,0 +1,167 @@
+        preferencesService.getSidePanePreferences().setPreferredPositions(preferredPositions);
+    }
+
+    public ObservableList<SidePaneType> getVisiblePanes() {
+        return visiblePanes;
+    }
+
+    /**
+     * @return True if <b>sidePane</b> is visible, and it can still move up therefore we should update the view
+     */
+    public boolean moveUp(SidePaneType sidePane) {
+        if (visiblePanes.contains(sidePane)) {
+            int currentPosition = visiblePanes.indexOf(sidePane);
+            if (currentPosition > 0) {
+                int newPosition = currentPosition - 1;
+                swap(visiblePanes, currentPosition, newPosition);
+                updatePreferredPositions();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return True if <b>sidePane</b> is visible, and it can still move down therefore we should update the view
+     */
+    public boolean moveDown(SidePaneType sidePane) {
+        if (visiblePanes.contains(sidePane)) {
+            int currentPosition = visiblePanes.indexOf(sidePane);
+            if (currentPosition < (visiblePanes.size() - 1)) {
+                int newPosition = currentPosition + 1;
+                swap(visiblePanes, currentPosition, newPosition);
+                updatePreferredPositions();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return True if <b>sidePane</b> is not already shown which means the view needs to be updated
+     */
+    public boolean show(SidePaneType sidePane) {
+        if (!visiblePanes.contains(sidePane)) {
+            visiblePanes.add(sidePane);
+            preferencesService.getSidePanePreferences().visiblePanes().add(sidePane);
+            visiblePanes.sorted(new PreferredIndexSort(preferencesService));
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @return True if <b>sidePane</b> is visible which means the view needs to be updated
+     */
+    public boolean hide(SidePaneType sidePane) {
+        if (visiblePanes.contains(sidePane)) {
+            visiblePanes.remove(sidePane);
+            preferencesService.getSidePanePreferences().visiblePanes().remove(sidePane);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public BooleanProperty groupsSidePaneVisibleProperty() {
+        return groupsPaneVisible;
+    }
+
+    public BooleanProperty openOfficePaneVisibleProperty() {
+        return openOfficePaneVisible;
+    }
+
+    public BooleanProperty webSearchSidePaneVisibleProperty() {
+        return webSearchPaneVisible;
+    }
+
+    public boolean isSidePaneVisible(SidePaneType sidePane) {
+        return visiblePanes.contains(sidePane);
+    }
+
+    /**
+     * This implementation is inefficient because of some JavaFX limitations, we only advice to use it on small lists
+     */
```

Comment:
Whom is it you are advising to?

Reply:
I thought in case someone wanted to share that functionality by moving it to an utility class for example, but now that I think about it, it's only 3 lines of code, they should figure that out without a comment :)